lib/api_auth/uri_header.ex

```diff
@@ -20,8 +20,16 @@ defmodule ApiAuth.UriHeader do
   end
 
   def parse_uri(uri) do
-    %{path: path} = URI.parse(uri)
+    %{path: path, query: query} = URI.parse(uri)
 
+    case query do
```

Comment:
I think this `case` statement can be replaced with `"#{value_for(path)}#{query}"`.

Reply:
You could have `"#{value_for(path)}?#{query}"` (note the `?`); however that would append the `?` to paths that don't have query params. I believe you will have to conditionally determine if there is a query string to know whether or not to include it, as done here with the case statement.

Let me know if you were thinking of something different, or if there's another control structure you'd like to use to conditionally add the `?`.

Reply:
It looks like I missed the `?` in `"#{path}?#{query}"`. I think the code here makes sense then.